### ISSUE TITLE
ESLint codeActionsOnSave not work

### ISSUE THREAD

**Issue by recallwei** [microsoft/vscode#194978]
Since I updated VSCode to v1.83.0, ESLint code actions on save does not work any more, and my user settings like below:
```json
{
  "editor.codeActionsOnSave": {
    "source.fixAll": "never",
    "source.fixAll.eslint": "always",
    "source.fixAll.stylelint": "never",
    "source.organizeImports": "never"
  }
}
```

**Comment by justschen** [comment]
PR is here: https://github.com/microsoft/vscode/pull/194930. Fixes an issue regarding a niche use case of code actions using arrays. As I mentioned, some code that I had been using for migration/conversion was removed and that unintentionally fixes this issue.

**Comment by recallwei** [comment]
{
  "editor.codeActionsOnSave": {
    "source.fixAll": false,
    "source.fixAll.eslint": true,
  }
}
This does not work for me after updating to v1.83.0.

**Comment by dbaeumer** [comment]
@recallwei I add a typo in my comment. A setting like the one you provided doesn't work right now. What should work as a work around is

```
{
  "editor.codeActionsOnSave": {
    "source.fixAll": true,
    "source.fixAll.eslint": true,
  }
}
```

**Comment by justschen** [comment]
also works for now is 

``` typescript
{
  "editor.codeActionsOnSave": {
    "source.fixAll": "always",
    "source.fixAll.eslint": "always",
  }
}
```

**Comment by justschen** [comment]
fix in PR now has intended behavior:
![Recording 2023-10-06 at 01 32 46](https://github.com/microsoft/vscode/assets/54879025/006dd82a-a0b5-4c2f-8fd6-a2190b490bab)

![Recording 2023-10-06 at 01 31 37](https://github.com/microsoft/vscode/assets/54879025/a772b2f0-b1a4-4030-bf18-0afcacb4348a)

**Comment by lianchi** [comment]
I am having the same issue since updating to 1.83.0.

My previous settings is like (NOT work):
```
"editor.codeActionsOnSave": {
  "source.fixAll": false,
  "source.fixAll.eslint": true
}
```

When I change it to:
```
"editor.codeActionsOnSave": {
  "source.fixAll": true,
  "source.fixAll.eslint": true
}
```
It works.

**Comment by u3u** [comment]
Whether setting `source.fixAll` to `true` or `always`, it no longer works for me. I can only manually execute the command `ESLint: Fix all auto-fixable Problems` 🙁

**Comment by kferch-teamworks** [comment]
Arrays for codeactionsonsave were what allowed the "Format Code Action", rohit-gohri.format-code-action to work. This was important because the array determines the order which you can't do with an object. This was broken with 1.83 change to object.

**Comment by justschen** [comment]
@kferch-teamworks related to https://github.com/microsoft/vscode/issues/194861, duplicate comment.

**Comment by liuliangsir** [comment]
That issue above seems to have disappeared after updating to `1.84.0-insider`. The information about system and setting look like the following.

```text
Version: 1.84.0-insider (user setup)
Commit: eb4f3c149f34d82b9763a08230907cf79e7fc103
Date: 2023-10-10T07:26:46.161Z
Electron: 25.8.4
ElectronBuildId: 24154031
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.19045
```

```json
{
  "editor.codeActionsOnSave": {
    "source.fixAll": "never",
    "source.fixAll.eslint": "explicit"
  },
}
```

**Comment by justschen** [comment]
closing! fix is out in release 1.83.2!

**Comment by mjfwebb** [comment]
> closing! fix is out in release 1.83.2!

Just for clarification did you mean 1.83.1?
